src/audio/eq_fir/eq_fir.c

```diff
@@ -461,13 +461,13 @@ static int eq_fir_cmd(struct comp_dev *dev, int cmd, void *data,
 
 static int eq_fir_trigger(struct comp_dev *dev, int cmd)
 {
-	struct comp_data *cd = comp_get_drvdata(dev);
+	struct comp_data *cd __unused = comp_get_drvdata(dev);
 
 	comp_info(dev, "eq_fir_trigger()");
 
-	if (cmd == COMP_TRIGGER_START || cmd == COMP_TRIGGER_RELEASE)
-		assert(cd->eq_fir_func);
-
+	if (cmd == COMP_TRIGGER_START || cmd == COMP_TRIGGER_RELEASE) {
+		; assert(cd->eq_fir_func);
+	}
```

Comment:
@singalsu should these be changed to unconditional `panic`s?

Reply:
Could this fail without panic by with -EINVAL returned if eq_fir_func is NULL? Comp_set_state() seems to return that with errors.

Reply:
Sure I will make it tomorrow. Thanks for spotting these @marc-hb !

Reply:
@marc-hb The PR is #5771